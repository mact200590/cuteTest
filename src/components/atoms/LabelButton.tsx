/* eslint-disable react/require-default-props */
import React from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  StyleProp,
  TextStyle,
  ViewStyle,
} from 'react-native';
import Label from './Label';

interface Props {
  onPress: () => void;
  label: string;
  fontStyle?: StyleProp<TextStyle>;
  fontStyleContainer?: StyleProp<ViewStyle>;
  disabled?: boolean;
}

const LabelButton = React.memo(
  ({
    onPress,
    label,
    fontStyle = {},
    fontStyleContainer = {},
    disabled = false,
  }: Props) => {
    return (
      <TouchableOpacity
        disabled={disabled}
        style={[styles.container, fontStyleContainer]}
        onPress={onPress}>
        <Label style={[styles.container, fontStyle]}>{label}</Label>
      </TouchableOpacity>
    );
  },
);

const styles = StyleSheet.create({
  container: {
    width: '100%',
    padding: 8,
  },
});

export default LabelButton;
