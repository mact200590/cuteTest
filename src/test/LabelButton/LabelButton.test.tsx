/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/react-native';
import LabelButton from '../../components/atoms/LabelButton';

describe('Test in <LabelButton/> component', () => {
  const handleOnPressMock = jest.fn();
  const dataMock = {
    label: 'text of button',
  };
  test('To toMatchSnapshot', () => {
    const tree = renderer
      .create(
        <LabelButton onPress={handleOnPressMock} label={dataMock.label} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
