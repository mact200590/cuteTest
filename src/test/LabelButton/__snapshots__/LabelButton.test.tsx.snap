// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <LabelButton/> component To toMatchSnapshot 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "padding": 8,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#6200ee",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        Array [
          Object {
            "padding": 8,
            "width": "100%",
          },
          Object {},
        ],
      ]
    }
  >
    text of button
  </Text>
</View>
`;
